base/bootstrap.go

```diff
@@ -282,6 +285,31 @@ func (cc *CouchbaseCluster) GetConfigBuckets() ([]string, error) {
 	return bucketList, nil
 }
 
+func (cc *CouchbaseCluster) GetClusterUUID(ctx context.Context, spec BucketSpec) (string, error) {
+	// choose random cached bucket connection
+	lits, err := cc.GetConfigBuckets()
+	if err != nil {
+		return "", err
+	}
+	randBucket := lits[rand.Intn(len(lits))]
+	b, teardown, err := cc.getBucket(ctx, randBucket)
+	if err != nil {
+		return "", err
+	}
+	defer teardown()
+
+	// construct what's needed for GocbV2Bucket to fetch cluster UUID
```

Comment:
Just move this function https://github.com/couchbase/sync_gateway/blob/main/db/database.go#L344 to base package?

Reply:
But we have an agent here, so I think maybe moving the function to not require a bucket would be great. It just needs a gocbcore.Agent to get mgmt endpoint

 https://github.com/couchbase/sync_gateway/blob/8a9c7f27bf4b527199552eb9466902568905ccdd/base/collection.go#L450
 
 Then you can have a way to get mgmteps from either couchbase cluster or bucket.